src/Pelagos/Bundle/AppBundle/Rabbit/Consumer/DoiConsumer.php

```diff
@@ -137,42 +137,41 @@ protected function issueDoi(Dataset $dataset, array $loggingContext)
 
         $issueMsg = ConsumerInterface::MSG_ACK;
         if ($this->doiAlreadyExists($dataset, $loggingContext)) {
-            return $this->updateDoi($dataset, $loggingContext);
-        }
-
-
-        $doiUtil = new DOIutil();
-
-        $generatedDOI = $doiUtil->generateDoi();
-
-        try {
-            $doiUtil->createDOI(
-                $generatedDOI,
-                'https://data.gulfresearchinitiative.org/tombstone/' . $dataset->getUdi(),
-                $dataset->getAuthors(),
-                $dataset->getTitle(),
-                $dataset->getReferenceDateYear(),
-                'Harte Research Institute'
-            );
-
-            $doi = new DOI($generatedDOI);
-            $doi->setCreator($dataset->getModifier());
-            $doi->setModifier($dataset->getModifier());
-            $dataset->setDoi($doi);
+            $this->logger->info('DOI Already issued for this dataset', $loggingContext);
```

Comment:
Shouldn't we at least do an update for the DOI?

Reply:
the last change what i made was exactly that, which led to this bug

Reply:
We run update on a separate request after issue I think anyway. So that should work.